File: event-schemas/src/entities/Asset.ts
import AssetType from './AssetType';
import AssetVariable from './AssetVariable';

class Asset {
  private _assetId: string;
  private _typeList: Array<AssetType>;
  private _inputs?: Array<string>;
  private _outputs?: Array<string>;
  private _classification?: string;
  private _categories?: Array<string>;
  private _variables?: Array<AssetVariable>;
  private _additionalProperties?: Map<String, object | string | number | Array<unknown> | boolean | null>;

  constructor(input: {
    assetId: string,
    typeList: Array<AssetType>,
    inputs?: Array<string>,
    outputs?: Array<string>,
    classification?: string,
    categories?: Array<string>,
    variables?: Array<AssetVariable>,
  }) {
    this._assetId = input.assetId;
    this._typeList = input.typeList;
    this._inputs = input.inputs;
    this._outputs = input.outputs;
    this._classification = input.classification;
    this._categories = input.categories;
    this._variables = input.variables;
  }

  get assetId(): string { return this._assetId; }
  set assetId(assetId: string) { this._assetId = assetId; }

  get typeList(): Array<AssetType> { return this._typeList; }
  set typeList(typeList: Array<AssetType>) { this._typeList = typeList; }

  get inputs(): Array<string> | undefined { return this._inputs; }
  set inputs(inputs: Array<string> | undefined) { this._inputs = inputs; }

  get outputs(): Array<string> | undefined { return this._outputs; }
  set outputs(outputs: Array<string> | undefined) { this._outputs = outputs; }

  get classification(): string | undefined { return this._classification; }
  set classification(classification: string | undefined) { this._classification = classification; }

  get categories(): Array<string> | undefined { return this._categories; }
  set categories(categories: Array<string> | undefined) { this._categories = categories; }

  get variables(): Array<AssetVariable> | undefined { return this._variables; }
  set variables(variables: Array<AssetVariable> | undefined) { this._variables = variables; }

  get additionalProperties(): Map<String, object | string | number | Array<unknown> | boolean | null> | undefined { return this._additionalProperties; }
  set additionalProperties(additionalProperties: Map<String, object | string | number | Array<unknown> | boolean | null> | undefined) { this._additionalProperties = additionalProperties; }
}
export default Asset;

File: event-schemas/data/csharp/Asset.cs
namespace entities
{
  using System.Collections.Generic;

  public class Asset
  {
    private string assetId;
    private AssetType[] typeList;
    private string[] inputs;
    private string[] outputs;
    private string classification;
    private string[] categories;
    private AssetVariable[] variables;
    private Dictionary<string, dynamic> additionalProperties;

    public string AssetId 
    {
      get { return assetId; }
      set { assetId = value; }
    }

    public AssetType[] TypeList 
    {
      get { return typeList; }
      set { typeList = value; }
    }

    public string[] Inputs 
    {
      get { return inputs; }
      set { inputs = value; }
    }

    public string[] Outputs 
    {
      get { return outputs; }
      set { outputs = value; }
    }

    public string Classification 
    {
      get { return classification; }
      set { classification = value; }
    }

    public string[] Categories 
    {
      get { return categories; }
      set { categories = value; }
    }

    public AssetVariable[] Variables 
    {
      get { return variables; }
      set { variables = value; }
    }

    public Dictionary<string, dynamic> AdditionalProperties 
    {
      get { return additionalProperties; }
      set { additionalProperties = value; }
    }
  }
}